src/backend/commands/tablecmds.c

```diff
@@ -826,6 +826,12 @@ DefineRelation(CreateStmt *stmt, char relkind, Oid ownerId,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("only shared relations can be placed in pg_global tablespace")));
 
+
+	if (pltsql_get_object_owner_hook && !OidIsValid(ownerId))
+	{
+		ownerId = (*pltsql_get_object_owner_hook)(namespaceId, ownerId);
+	}
+
 	/* Identify user ID that will own the table */
```

Comment:
For CREATEs, Do we need to handle TYPE object?

Reply:
Done